system/src/commonMain/kotlin/com/splendo/kaluga/system/network/BaseNetworkManager.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+abstract class BaseNetworkManager(private val networkStateRepo: NetworkStateRepo) : CoroutineScope by networkStateRepo {
+
+    internal abstract suspend fun startMonitoringNetwork()
+    internal abstract suspend fun stopMonitoringNetwork()
+    internal abstract suspend fun isNetworkEnabled(): Boolean
```

Comment:
Can be read from state instead of this method (this method probably returns something wrong when still initializing). Can make an extension method that does `flow().first()` to shortcut this.

Reply:
During the initialization it would return false since it is initialValue for the `AtomicReference` in the implementation.

Reply:
Do you mean an extension function `NetworkStateRepo`?

> Can make an extension method that does flow().first() to shortcut this.

Reply:
yes, but I realize now this is not even for end users. It can just be removed after your other changes I think.

For end users this could still be a nice addition in StateRepo of course (idea stolen from @Daeda88 in location).

Reply:
853749a